### PR TITLE
add installing results with external database docs

### LINKED ISSUE #371: Add instructions on setting up an external Postgres database
Improve the documentation by adding instructions on how users can connect the API server to an external Postgres database.

### PR DESCRIPTION
**Pull request by avinal**
# Changes

- Results release contains database by default
- Use patches to remove database component from default install
- fixes #371 

 /kind documentation
# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: docs/external-database.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+above and add them to `kustomization.yaml` instead of patches:
+
+```yaml
+patches:
+  - path: delete-database-statefulset.yaml
+  - path: delete-database-service.yaml
+  - path: delete-database-configmap.yaml
+```
+
+## Modifying DB_HOST and DB_NAME
```
Comment (adambkaplan) [inline docs/external-database.md:68]: Don't we need another section describing how to set the remote database username and password?

File: docs/external-database.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+You can use any external Postgres database instead of a local database deployment.
+Tekton Results gets its database configuration either from a Kubernetes Secret
+named `tekton-results-postgres` or from a ConfigMap with the same name.
+
+Tekton Results expects following database configurations to be provided.
+
+| Name | Secret | ConfigMap | Comments |
+| --- | --- | --- | --- |
+| Database User | `POSTGRES_USER` | `DB_USER` | default: postgres |
+| Database Password | `POSTGRES_PASSWORD` | `DB_PASSWORD` | |
```
Comment (adambkaplan) [inline docs/external-database.md:12]: We should not recommend users provide sensitive database credentials in ConfigMaps.

**Review by adambkaplan — APPROVED** [review]
/approve

Overall this is great! I just called out two items related to database credentials.

File: docs/external-database.md (hunk 1 above)
Comment (avinal) [reply docs/external-database.md:68]: Ah yes, I missed it. I will add it.

**Comment by khrm** [comment]
@avinal Please rebase this.

**Review by khrm — COMMENTED** [review]
/lgtm

**Comment by khrm** [comment]
Seems tide was incorrectly requesting to do the rebase.

**Comment by avinal** [comment]
I believe, it rebases in most cases automatically if needed.

**Pull request merged**